packages/patternfly-4/react-core/src/components/Page/Page.tsx

```diff
@@ -1,56 +1,52 @@
-import React, { cloneElement } from 'react';
-import PropTypes from 'prop-types';
+import * as React from 'react';
 import styles from '@patternfly/react-styles/css/components/Page/page';
 import { css } from '@patternfly/react-styles';
 import { global_breakpoint_md as globalBreakpointMd } from '@patternfly/react-tokens';
 import { debounce } from '../../helpers/util';
 
-export const PageLayouts = {
-  vertical: 'vertical',
-  horizontal: 'horizontal'
+export enum PageLayouts {
+  vertical = 'vertical',
+  horizontal = 'horizontal'
 };
 
-const propTypes = {
-  /** Content rendered inside the main section of the page layout (e.g. <PageSection />) */
-  children: PropTypes.node,
-  /** Additional classes added to the page layout */
-  className: PropTypes.string,
-  /** Header component (e.g. <PageHeader />) */
-  header: PropTypes.node,
-  /** Sidebar component for a side nav (e.g. <PageSidebar />) */
-  sidebar: PropTypes.node,
-  /** Skip to content component for the page */
-  skipToContent: PropTypes.node,
-  /** Breadcrumb component for the page */
-  breadcrumb: PropTypes.node,
-  /** If true, manages the sidebar open/close state and there is no need to pass the isNavOpen boolean into
-   * the sidebar component or add a callback onNavToggle function into the PageHeader component */
-  isManagedSidebar: PropTypes.bool,
-  /** Can add callback to be notified when resize occurs, for example to set the sidebar isNav prop to false for a width < 768px
-   * Returns object { mobileView: boolean, windowSize: number } */
-  onPageResize: PropTypes.func,
-  /** Additional props are spread to the container <div> */
-  '': PropTypes.any // eslint-disable-line react/require-default-props
-};
+export interface PageProps extends React.HTMLProps<HTMLDivElement> {
+  children?: React.ReactNode;
+  className?: string;
+  header?: React.ReactNode;
+  sidebar?: React.ReactNode;
+  skipToContent?: React.ReactElement;
+  isManagedSidebar?: boolean;
+  onPageResize?: (object: object) => void;
+  breadcrumb?: React.ReactNode;
+}
 
-const defaultProps = {
-  breadcrumb: null,
-  children: null,
-  className: '',
-  header: null,
-  sidebar: null,
-  skipToContent: null,
-  isManagedSidebar: false,
-  onPageResize: null
-};
+export interface PageState {
```

Comment:
You don't have to export it as an interface like this. You can declare it as a class member.

Reply:
but I use it in the class's generics, it can also be a class member? 
```
export class Page extends React.Component<PageProps, PageState> {
```
Can you pls send a snippet example?  -thanks

Reply:
...or are you saying just don't export it?  :-)

Reply:
All you need is this:
```js
 23 export class Page extends React.Component<PageProps> { 
 24   state = { 
 25     desktopIsNavOpen: true,
 26     mobileIsNavOpen: false,
 27     mobileView: false
 28   };
```
and you can remove `PageState`.

Reply:
Ok, but I already set state to default values in the constructor.  PageState was to typescript the state object properties.

Reply:
Yes, I know. Move them out of the constructor and into a class member like above and it's the same thing and TypeScript sees the `state` class attribute and infers the types from there. End result less code and a little more elegance :)

Reply:
Ok, fixed, thanks